kafka/consumer-group.go

```diff
@@ -0,0 +1,81 @@
+package kafka
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/Shopify/sarama"
+	"github.com/bsm/sarama-cluster"
+)
+
+var tick = time.Millisecond * 4000
+
+type ConsumerGroup struct {
+	Consumer *cluster.Consumer
+	Incoming chan Message
+	Closer   chan bool
+	Errors   chan error
```

Comment:
In the observation extractor I ended up adding methods to return 'Incoming' and 'Closer' and making the struct fields private. This enabled me to create an interface like this 

```
// MessageConsumer provides a generic interface for consuming []byte messages
type MessageConsumer interface {
	Incoming() chan kafka.Message
	Closer() chan bool
}
```
I could then easily test around Kafka using a mock consumer: https://github.com/ONSdigital/dp-observation-extractor/blob/feature/initial-implementation/event/eventtest/message_consumer.go

Do you think its worth having the struct members private with methods to access them? or is there another way of mocking them that you can show me? (this also applied to the producer)